lib/geo_record.rb

```diff
@@ -1,4 +1,19 @@
+require "delegate"
+
 module GeoRecord
+  # Ensure that when coordinates are printed that they are always in decimal degrees,
+  # and not e.g. 4.0e-05
+  # Unfortunately you can't extend Numeric classes directly (e.g. `Coord < Float`).
+  class Coord < DelegateClass(Float)
```

Comment:
What's the benefit of using delegation here over just inheriting from `Float` and overriding the `to_s` method?

If we did that then I think we wouldn't need to special case the AMF encoding (because Coord would match Numeric) and we wouldn't need `to_f` on the pack etc?